openapi-cli/src/main/java/io/ballerina/openapi/cmd/BallerinaCodeGenerator.java

```diff
@@ -409,6 +417,11 @@ public List<GenSrcFile> generateBallerinaService(Path openAPI, String serviceNam
                     openAPI);
         }
 
+        if (SUPPORTED_OPENAPI_VERSIONS.contains(openAPIDef.getOpenapi())) {
+            outStream.printf("WARNING: The tool has not been tested with OpenAPI version %s. " +
+                    "The generated code may potentially contain errors.%n", openAPIDef.getOpenapi());
+        }
```

Comment:
Shall we move this check to a util method and reuse instead of having in multiple places?

Reply:
Addressed in f29a746